blocks/library/table2/table-block.js

```diff
@@ -0,0 +1,85 @@
+import Editable from '../../editable';
+import BlockMenu from '../../block-menu';
+
+function execCommand( command ) {
+	return ( editor ) => {
+		if ( editor ) {
+			editor.execCommand( command );
+		}
+	};
+}
+
+const TABLE_CONTROLS = [
+	{
+		icon: 'table-row-before',
+		title: wp.i18n.__( 'Insert Row Before' ),
+		onClick: execCommand( 'mceTableInsertRowBefore' ),
+	},
+	{
+		icon: 'table-row-after',
+		title: wp.i18n.__( 'Insert Row After' ),
+		onClick: execCommand( 'mceTableInsertRowAfter' ),
+	},
+	{
+		icon: 'table-row-delete',
+		title: wp.i18n.__( 'Delete Row' ),
+		onClick: execCommand( 'mceTableDeleteRow' ),
+	},
+	{
+		icon: 'table-col-before',
+		title: wp.i18n.__( 'Insert Column Before' ),
+		onClick: execCommand( 'mceTableInsertColBefore' ),
+	},
+	{
+		icon: 'table-col-after',
+		title: wp.i18n.__( 'Insert Column After' ),
+		onClick: execCommand( 'mceTableInsertColAfter' ),
+	},
+	{
+		icon: 'table-col-delete',
+		title: wp.i18n.__( 'Delete Column' ),
+		onClick: execCommand( 'mceTableDeleteCol' ),
+	},
+];
+
+export default class TableBlock extends wp.element.Component {
+	constructor() {
+		super();
+		this.state = {
+			editor: null,
+		};
+	}
+
+	render() {
+		const { content, focus, onFocus, onChange } = this.props;
+
+		return [
+			focus && (
+				<BlockMenu
```

Comment:
With the toolbar-menu component I was trying to replicate the mockups that @mapk and @jasmussen were creating ( https://github.com/WordPress/gutenberg/issues/147#issuecomment-294638360 ). It was basically a generalization of the block switcher component with the intent that maybe the block switcher could be changed to share code with it.

At the moment the BlockMenu component is a thin wrapper around it but I admit that it has problems. I am learning react as I go (it's not used at Ephox) so I am not very knowledgeable about the slot/fill mechanic. What we probably want to happen is that for multiple uses of the BlockMenu it will add menu items to the menu rather than creating a new menu component each time.